src/tetris/player/localPlayer.ts

```diff
@@ -0,0 +1,68 @@
+/// <reference path="../../../definitions/phaser.d.ts"/>
+
+import Field from 'tetris/field/field';
+import Player from "tetris/player/player";
+import BiasEventReceiver from 'tetris/biasEngine/biasEventReceiver';
+import KeyboardManager = Phaser.Input.Keyboard.KeyboardManager;
+import BiasEvenType from 'tetris/biasEngine/biasEventType';
+import BiasEventDuplicateInput from 'tetris/biasEngine/events/biasEventDuplicateInput';
+
+export default class LocalPlayer extends Player {
+
+	//region public members
+	//endregion
+
+	//region public methods
+	public update(time: number, delta: number): void {
+		// update bias events
+		this._biasEventReceiver.update(time, delta);
+		// apply bias
+		if (this._biasEventReceiver.has(BiasEvenType.DisableInput)) {
+			return;
+		}
+		let moveOperation: () => void;
+
+		// handle keyboard inputs
+		if(this._cursor.left.isDown) {
+			moveOperation = this.moveLeft;
+		} else if (this._cursor.right.isDown) {
+			moveOperation = this.moveRight;
+		} else if (this._cursor.down.isDown) {
+			moveOperation = this.moveDown;
+		} else if (this._cursor.space.isDown) {
+			moveOperation = this.drop;
+		} else if (this._cursor.up.isDown) {
+			moveOperation = this.rotate;
+		}
+
+		moveOperation();
+
+		// check whether to apply operation a second time (as bias)
+		if (this._biasEventReceiver.has(BiasEvenType.DuplicateInput)) {
```

Comment:
Please move to an extra method `_applyDuplicateInputBias(moveOperation)` for better readability.